### PR TITLE
lowrisc: flash_ctrl: Convert erase_page to be async

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

Convert erase_page to be async

### Testing Strategy

Running on OpenTitan.

### TODO or Help Wanted

N/A

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Review by bradjc — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]

File: chips/lowrisc/src/flash_ctrl.rs [hunk 1]
```diff
@@ -340,34 +340,49 @@ impl<'a> FlashCtrl<'a> {
...
-                if self.read_index.get() >= buf.0.len() {
-                    // We have all of the data, call the client
-                    self.flash_client.map(move |client| {
-                        client.read_complete(buf, hil::flash::Error::CommandComplete);
-                    });
-                } else {
-                    // Still waiting on data, keep waiting
-                    self.read_buf.replace(buf);
-                    self.enable_interrupts();
+            if self.registers.control.read(CONTROL::OP) == (u32::from(CONTROL::OP::READ) >> 4) {
```
[the remark this answers is not in the document]
Comment (alistair23) [reply chips/lowrisc/src/flash_ctrl.rs]: Ah, I knew there had to be a better way.

**Review by bradjc — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]
bors d+

From just this diff alone, it's a bit weird that we used to have a call to `erase_page` and now we don't, but I'm assuming that was an artifact of the old implementation and simply not needed any more. That said, I'll leave to @alistair23 to merge just in case it needs to be put back somewhere.

**Comment by alistair23** [comment]
Thanks Pat.

That was actually a mistake in the first implementation. Flash always needs to be erased before it can be written so I previously did that in the OT driver. What should actually happen is the flash user should call erase first, which is fixed here.

**Pull request merged**